cloudsql/mysql/database-sql/cloudsql.go

```diff
@@ -0,0 +1,299 @@
+			log.Fatalf("indexHandler: error returned from showTotals: %s", err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+	case "POST":
+		saveVote(w, r)
+	default:
+		http.Error(w, fmt.Sprintf("HTTP Method %s Not Allowed", r.Method), http.StatusMethodNotAllowed)
+	}
+}
+
+// recentVotes returns a slice of the last 5 votes cast.
+func recentVotes() ([]vote, error) {
+	var votes []vote
+	rows, err := db.Query(`SELECT candidate, time_cast FROM votes ORDER BY time_cast DESC LIMIT 5`)
+	if err != nil {
+		return votes, err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		nextVote := vote{}
+		rowError := rows.Scan(&nextVote.Candidate, &nextVote.VoteTime)
+		if err != nil {
+			return votes, rowError
+		}
+		votes = append(votes, nextVote)
+	}
+	if err = rows.Err(); err != nil {
+		return votes, err
+	}
+	return votes, nil
+}
+
+// currentTotals returns a templateData structure for populating the web page.
+func currentTotals() (templateData, error) {
+
+	// get total votes for each candidate
+	var tabVotes, spaceVotes uint
+	err := db.QueryRow(`SELECT count(vote_id) FROM votes WHERE candidate='TABS'`).Scan(&tabVotes)
+	if err != nil {
+		return templateData{}, err
+	}
+	err = db.QueryRow(`SELECT count(vote_id) FROM votes WHERE candidate='SPACES'`).Scan(&spaceVotes)
+	if err != nil {
+		return templateData{}, err
+	}
+
+	var voteDiffStr string = voteDiff(int(math.Abs(float64(tabVotes) - float64(spaceVotes)))).String()
+
+	latestVotesCast, err := recentVotes()
+	if err != nil {
+		return templateData{}, err
+	}
+	return templateData{tabVotes, spaceVotes, voteDiffStr, latestVotesCast}, nil
+
+}
+
+// showTotals renders an HTML template showing the current vote totals.
+func showTotals(w http.ResponseWriter, r *http.Request) error {
+
+	totals, err := currentTotals()
+	if err != nil {
+		return err
+	}
+	err = parsedTemplate.Execute(w, totals)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// saveVote saves a vote passed as http.Request form data.
+func saveVote(w http.ResponseWriter, r *http.Request) error {
+	if err := r.ParseForm(); err != nil {
+		return err
+	}
+
+	var team string
+	if teamprop, ok := r.Form["team"]; ok {
+		team = teamprop[0]
+	} else {
+		return errors.New("saveVote: team property missing from form response")
+	}
+
+	// [START cloud_sql_mysql_databasesql_connection]
+	sqlInsert := "INSERT INTO votes (candidate) VALUES (?)"
+	if team == "TABS" || team == "SPACES" {
+		if _, err := db.Exec(sqlInsert, team); err != nil {
+			fmt.Fprintf(w, "unable to save vote: %s", err)
+			return err
+		} else {
+			fmt.Fprintf(w, "Vote successfully cast for %s!\n", team)
+		}
+	}
+	return nil
+	// [END cloud_sql_mysql_databasesql_connection]
+}
+
+// mustGetEnv is a helper function for getting environment variables.
+// Displays a warning if the environment variable is not set.
+func mustGetenv(k string) string {
+	v := os.Getenv(k)
+	if v == "" {
+		fmt.Printf("Warning: %s environment variable not set.\n", k)
```

Comment:
While `fmt.Println` works to write to stdout, once you start using log it's best to use it consistently. Also helps readability as it makes an easier distinction from `fmt.Fprintf`.

```suggestion
		log.Printf("Warning: %s environment variable not set.\n", k)
```